File: algorithms/recursions/cc/sum_first_n_number.cc
#include <iostream>
using namespace std;


void sumOfFirstNNumber(int num, int sum) {
  if(num < 1) {
    cout << sum << endl;
    return;
  }
  sumOfFirstNNumber(num - 1, sum + num);
}
int main() {
  sumOfFirstNNumber(5, 0);
  return 0;
}

File: algorithms/recursions/ts/sum_first_n_number.ts
// function sumFirstNNumber(num: number, sum: number):void {
//   if(num < 1) {
//     console.log("Sum is: ", sum);
//     return;
//   }
//   sumFirstNNumber(num -1, sum + num);
// }

function sumFirstNNumber(num: number): number {
  if (num < 1) {
    return 0;
  }
  return num + sumFirstNNumber(num - 1);
}
// sumFirstNNumber(5, 0);
const sum = sumFirstNNumber(15);
console.log(sum);
